skbio/alignment/tests/test_tabular_msa.py

```diff
@@ -562,7 +630,7 @@ def test_reassign_index_with_mapping_callable(self):
         seqs = [DNA("A"), DNA("C"), DNA("G")]
 
         msa = TabularMSA(seqs, index=[0, 1, 2])
-        msa.reassign_index(mapping=str)
+        msa.reassign_index(mapping=lambda e: str(e))
```

Comment:
This was a backwards-incompatible change to `pd.Series.rename`. Passing `mapping=str` now modifies the series' `name` attribute (feature? bug? [the docs](http://pandas.pydata.org/pandas-docs/stable/generated/pandas.Series.rename.html) aren't helpful).

Do you think we should continue using `Series.rename` as a passthrough for `mapping`, or roll our own?